pyacq/core/tools.py

```diff
@@ -52,13 +52,22 @@ def process_data(self, pos, data):
         self.new_data.emit(self._pos, data)
     
     def stop(self):
-        with self.lock:
+        with self.running_lock:
             self.running = False
     
     def pos(self):
         return self._pos
 
-
+class ThreadPollOuput(ThreadPollInput):
```

Comment:
Yep maybe the name is not good.
The idea is same as ThreadPollinput (that pool a client socket InputStream) but you give it an OuputStream (server socket) so it create internally it is own InputStream.
Maybe you could help with a good english to describe this.
the code uis more clear than the name.